test/notebooks/test_notebooks.py

```diff
@@ -0,0 +1,106 @@
+import glob
+import os
+import time
+import urllib.request
+from pathlib import Path
+from zipfile import ZipFile
+
+import nbformat
+import numpy as np
+import pandas as pd
+import papermill as pm
+import torch
+
+import syft as sy
+from syft import TorchHook
+from syft.workers.websocket_server import WebsocketServerWorker
+
+# buggy notebooks
+exclusion_list_basic = [
+    "Part 10 - Federated Learning with Secure Aggregation.ipynb",
+    "Part 13b - Secure Classification with Syft Keras and TFE - Secure Model Serving.ipynb",
+    "Part 13c - Secure Classification with Syft Keras and TFE - Private Prediction Client.ipynb",
+]
+
+exclusion_list_advanced = [
+    "Build your own tensor type (advanced).ipynb",
+    "Federated Recurrent Neural Network.ipynb",
+]
+
+
+def test_notebooks_basic(isolated_filesystem):
+    """Test Notebooks in the tutorial root folder."""
+    notebooks = glob.glob("*.ipynb")
+    for notebook in notebooks:
+        if Path(notebook).name in exclusion_list_basic:
+            continue
+        res = pm.execute_notebook(
+            notebook, "/dev/null", parameters={"epochs": 1, "n_test_batches": 5}, timeout=300
+        )
+        assert isinstance(res, nbformat.notebooknode.NotebookNode)
+
+
+def test_notebooks_advanced(isolated_filesystem):
```

Comment:
Why do you need to separate these? besides the path

Reply:
good question, I did initially because the tests where hard to debug and taking long. But I was thinking to work with a more general approach soon because that way new notebooks are discovered and don't have to be added manually.

Reply:
I though it's because you wanna give different timeout or so, but I do not think it's currently the case :)